### DISCUSSION TITLE
FluxCD with CodeCommit

### DISCUSSION THREAD

**Discussion by TechGirl-3** [fluxcd/flux2#808] [Q&A]
Hi,

I am new to FluxCD and I am looking for FluxCD integration with CodeCommit. 

Thanks.

**Comment by stefanprodan** [comment]
Have you tried using it and it didn't work?

**Reply by TechGirl-3** [reply]
I haven't tried it yet.  I was looking for any documentation available for FluxCD and AWS CodeCommit.

**Reply by stefanprodan** [reply]
There are no docs for it as we (the maintainers) don't use CodeCommit.

Using `flux create source git myrepo --url=ssh://<code-commit-repo>` should work if CodeCommit has support for SSH. Another option is to use HTTPS with a token e.g. `flux create source git myrepo --url=https://<code-commit-repo> --username=git --password=my-token`.

**Reply by TechGirl-3** [reply]
Thank you for your reply.

**Reply by phillebaba** [reply]
Just a tip, I would avoid trying to use IAM role authentication to code commit as it requires a credential helper to generate a temporary token. It will probably be difficult to integrate with fluxv2, instead you should create static user credentials which you can store in the cluster.

Docs related to what you should avoid.
https://docs.aws.amazon.com/codecommit/latest/userguide/setting-up-https-unixes.html

**Reply by rossigee** [reply]
@TechGirl-3 - ftr, I haven't been able to get it to work yet, and there seems to others having similar issues using CodeCommit with flux2...

https://github.com/fluxcd/flux2/discussions/948
